native/src/boot/bootimg.cpp

```diff
@@ -479,14 +501,29 @@ int unpack(const char *image, bool skip_decomp, bool hdr) {
     dump(boot.kernel_dtb, boot.hdr->kernel_dt_size, KER_DTB_FILE);
 
     // Dump ramdisk
-    if (!skip_decomp && COMPRESSED(boot.r_fmt)) {
-        if (boot.hdr->ramdisk_size() != 0) {
-            int fd = creat(RAMDISK_FILE, 0644);
-            decompress(boot.r_fmt, fd, boot.ramdisk, boot.hdr->ramdisk_size());
-            close(fd);
+    if (boot.hdr->ramdisk_size() != 0) {
+        for (int i = 0; i < boot.num_vendor_ramdisk_table_entries; ++i) {
+            std::string file_name(RAMDISK_FILE);
+            if (boot.hdr->header_version() == 4 && boot.hdr->is_vendor) {
+                file_name.assign("ramdisk.");
+                file_name.append(to_string(i));
+                file_name.append(".cpio");
+            }
```

Comment:
```suggestion
            char file_name[PATH_MAX] = {};
            ssprintf(file_name, sizeof(file_name), VENDOR_RAMDISK_FILE, i);
```

Reply:
Switched to ssprintf, but change file name only for vendor_boot v4 to keep backwards compatibility with scripts etc.